external-crates/move/crates/move-compiler/tests/linter/true_positive_excessive_nesting.move

```diff
@@ -0,0 +1,30 @@
+module 0x42::excessive_nesting_true_positive {
+
+    #[allow(unused_assignment)]
+    fun test_true_positive_direct_nesting() {
+        let value = 100;
+        
+        // Direct nesting that should be combined with &&
+        if (value > 50) {
+            if (value < 150) {  // Should trigger warning
+                if (value != 100) {  // Should trigger another warning
+                    value = 75;
+                };
+            };
+        };
+
+    }
+
+    #[allow(unused_assignment)]
+    fun test_true_positive_complex_nesting() {
+        let value = 100;
+        if (value > 0) {
+            let temp = value + 50;
+            if (temp > 100) {  // Should trigger warning - indirect nesting
```

Comment:
For clarity, this case should not trigger since it is not in a block of length 1